serverless.yml

```diff
@@ -0,0 +1,125 @@
+service: github-sso
+
+plugins:
+  - serverless-webpack
+
+package:
+  exclude:
+    - .git/*
+    - npm-debug*
+
+provider:
+  environment:
+    COGNITO_REDIRECT_URI: ${env:COGNITO_REDIRECT_URI, file(./config-dev.yml):COGNITO_REDIRECT_URI}
+    GITHUB_CLIENT_ID: ${self:custom.githubClientId}
+    GITHUB_CLIENT_SECRET: ${self:custom.githubClientSecret}
+    GITHUB_LOGIN_URL: "https://github.com"
+  iamRoleStatements:
+    - Effect: "Allow"
+      Action:
+        - "dynamodb:*"
+      Resource:
+        Fn::Join:
+          - ""
+          - - "arn:aws:dynamodb:"
+            - Ref: AWS::Region
+            - ":"
+            - Ref: AWS::AccountId
+            - ":"
+            - "table/${self:service}-${self:custom.stage}-*"
+  name: aws
+  region: us-east-1
+  runtime: nodejs10.x
+  stage: dev
+
+custom:
+  awsUserPoolId: ${env:AWS_USER_POOL_ID, file(./config-dev.yml):AWS_USER_POOL_ID}
+  awsGithubSsoGatewayUri:
+    Fn::Join:
+      - ""
+      - - "https://"
+        - Ref: ApiGatewayRestApi
+        - .execute-api.${self:provider.region}.amazonaws.com/${self:custom.stage}
+  githubClientId: ${env:GITHUB_CLIENT_ID, file(./config-dev.yml):GITHUB_CLIENT_ID}
+  githubClientSecret: ${env:GITHUB_CLIENT_SECRET, file(./config-dev.yml):GITHUB_CLIENT_SECRET}
+  lambda:
+    provisionedConcurrency:
+      production: 1
+      dev: 0
+  stage: ${opt:stage, self:provider.stage}
+  webpack:
+    webpackConfig: 'webpack.config.serverless.js'
+
+functions:
+  authorize:
+    handler: src/connectors/lambda/authorize.handler
+    events:
+      - http:
+          path: /authorize
+          method: get
+          cors: true
+    environment:
+      DYNAMODB_STATE_TABLE: !Ref CognitoStatesTable
+      REGION: !Ref "AWS::Region"
+    provisionedConcurrency: ${self:custom.lambda.provisionedConcurrency.${self:custom.stage}}
+  callback:
+    handler: src/connectors/lambda/callback.handler
+    events:
+      - http:
+          path: /callback
+          method: get
+          cors: true
+    environment:
+      DYNAMODB_STATE_TABLE: !Ref CognitoStatesTable
+      REGION: !Ref "AWS::Region"
```

Comment:
You can get rid of these refs by adding a serverless plugin (if you want): https://serverless.com/plugins/serverless-pseudo-parameters/

Reply:
good catch - this was a remanent. updated to use `${self:provider.region}` without the plugin.

Reply:
updated